addons/account/static/src/js/bills_upload.js

```diff
@@ -1,160 +1,170 @@
-            }),
-        }),
-    });
-
-    var DocumentsKanbanView = KanbanView.extend({
-        config: Object.assign({}, KanbanView.prototype.config, {
-            Controller: KanbanController.extend({_dropZone: '.o_content'}, UploadDocumentMixin, {
-                buttons_template: 'DocumentsKanbanView.buttons',
-                events: Object.assign({}, KanbanController.prototype.events, UploadDocumentMixin.events),
-            }),
-        }),
-    });
-
-    var DashboardKanbanController = KanbanController.extend({_dropZone: '.o_kanban_record'}, UploadDocumentMixin, {
-        events: Object.assign({}, KanbanController.prototype.events, UploadDocumentMixin.events),
-
-        _setJournalContext: function(ev) {
-            var kanbanRecord = $(ev.currentTarget).closest('.o_kanban_record').data('record');
-            this.initialState.context['default_journal_id'] = kanbanRecord.id;
-            let journal_type = kanbanRecord.state.data.type;
-            if (journal_type == 'sale') {
-                this.initialState.context['default_move_type'] = 'out_invoice';
-            } else if (journal_type == 'purchase') {
-                this.initialState.context['default_move_type'] = 'in_invoice';
-            }
-        },
-
-        _onUpload: function (ev) {
-            this._setJournalContext(ev);
-            UploadDocumentMixin._onUpload.apply(this, arguments);
-        },
-
-        _onDrop: function (ev) {
-            this._setJournalContext(ev);
-            UploadDocumentMixin._onDrop.apply(this, arguments);
-        },
-    });
-
-    var DashboardKanbanView = KanbanView.extend({
-        config: Object.assign({}, KanbanView.prototype.config, {
-            Controller: DashboardKanbanController,
-        }),
-    });
-
-    viewRegistry.add('account_tree', DocumentsListView);
-    viewRegistry.add('account_documents_kanban', DocumentsKanbanView);
-    viewRegistry.add('account_dashboard_kanban', DashboardKanbanView);
-});
+    async onFileUploaded(file) {
+        let att_data = {
+            name: file.name,
+            mimetype: file.type,
+            datas: file.data,
+        }
+        const att_id = await this.orm.call("ir.attachment", "create", [att_data], {
+            context: {...this.props.extraContext, ...this.env.searchModel.context},
+        });
+        this.attachment_ids_to_process.push(att_id);
+    }
+
+    async onUploadComplete() {
+        const action = await this.orm.call("account.journal", "create_document_from_attachment", ["", this.attachment_ids_to_process], {
+            context: {...this.props.extraContext, ...this.env.searchModel.context},
+        });
+        this.attachment_ids_to_process = [];
+        this.action.doAction(action);
+    }
+}
+AccountFileUploader.components = {
+    FileUploader,
+}
+AccountFileUploader.template = "account.HiddenFileUploader";
+
+class AccountDropZone extends Component {}
+AccountDropZone.defaultProps = {
+    visibilityClass: "d-none",
```

Comment:
Why are we using a class to hide the dropzone instead of just not rendering it with a t-if when it's not needed?
It also seems like this component has basically no behaviour and is just a template, so you could just t-call the template instead of using a component.

Reply:
if I use a `t-if` I get a "Component is destroyed" when using it on the kanbanRecord https://github.com/odoo/odoo/pull/96672/commits/5ee56d246f39d027053adb1434601668c8591d15#diff-f48f436b6c64a5ab4918a6f53fdb2584c42427fe29f56b7689d607c43e51481cR41